src/app/dao/tests/PartDAO.spec.ts

```diff
@@ -496,29 +529,34 @@ describe('PartDAO', () => {
             await signOut();
             const playerZero: MinimalUser = await createConnectedUser(CREATOR_EMAIL, CREATOR_NAME);
 
-            const turnDeltas: number[] = [
-                +1, // move
-                -1, // take back
-                -2, // take back when it was our turn again
+            const turnDeltasAndListMoves: [number, number[]][] = [
+                [+1, [2, 1, 0]], // move
+                [-1, [0]], // take back
+                [-2, []], // take back when it was our turn again
```

Comment:
tu met la liste des moves de droite à gauche oO ?
[0, 1] + un move [0, 1, 2]
[0, 1] - un move [0]
[0, 1] - 2 moves []

Reply:
aaah ça veut dire qu'on parcours toute la liste de mouvements À CHAQUE MOUVEMENT ? :D Une partie de 100 mouvement requière donc O(10000) opérations plutôt que O(100) si on le faisait dans le sense inverse :o

Reply:
Non, mais on lit juste le dernier élément, là dans ce que tu fais on applique 0 puis 0 :D !
Mais non fait ça:
```
while (rules.node.gameState.turn < listMoves.length) {
            const currentPartTurn: number = rules.node.gameState.turn;
            const chosenMove: Move = this.gameComponent.encoder.decode(listMoves[currentPartTurn]);
```

Reply:
changé pour left-to-right !